### PR TITLE
more docs

### PR DESCRIPTION
**Pull request by KristianKjerstad**
## Why is this pull request needed?

add more docs about using Git and setting up debugger

## What does this pull request change?

docs
## Issues related to this change:

### CODE REVIEW

File: docs/getting-started/inital-setup.md [hunk 1]
```diff
@@ -85,3 +85,34 @@ To let the IDE know that dependencies can be found in `.yarn` instead of `node_m
...
+4) Select Docker compose in the left menu and Choose the following settings:
+   1) Server: Docker
+   2) Configuration files: select docker-compose.yml and docker-compose.override.yml
+   3) Service: api
+   Click "OK" and "Apply"
+5) Click on "Add Configutarion..." in top right corner in PyCharm
+6) Click on + and select Python and enter the following settings
+   1) Script path: api/src/app.py
+   2) parameters: run
+   3) python interpreter: the newly created remove docker compose interpreter (step 4)
```
Comment (soofstad) [inline docs/getting-started/inital-setup.md:106]:
```suggestion
   3) python interpreter: the newly created docker compose interpreter (step 4)
```

File: docs/getting-started/prerequisites.md [hunk 2]
```diff
@@ -10,4 +10,5 @@ In order to run you need to have installed:
 - [Docker](https://www.docker.com/)
 - [Docker Compose](https://docs.docker.com/compose/)
 - Git
-- [Python](https://www.python.org/) (3.10 or newer)
\ No newline at end of file
+- [Python](https://www.python.org/) (3.10 or newer)
+- [NodeJS](https://nodejs.org/en/)
```
Comment (soofstad) [inline docs/getting-started/prerequisites.md:14]: Could put "python" and "node" under a separate "If you want to run locally".
Could then also add "poetry" and "yarn"

File: docs/how-to/using-git.md [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+BREAKING CHANGE: <description>
+```
+
+Also, remember to set up [Pre-commit](pre-commit.md).
+
+
+## What to check before you send a PR to code review
+* You use conventional commit messages
+* Unit tests passes
```
Comment (soofstad) [inline docs/how-to/using-git.md:59]:
```suggestion
* All tests passes
```

File: docs/how-to/using-git.md [hunk 4]
```diff
@@ -0,0 +1,70 @@
...
+* Other automatic tests passes, given it is feasible to execute them before code review
```
Comment (soofstad) [inline docs/how-to/using-git.md:60]:
```suggestion
```

File: docs/how-to/using-git.md [hunk 5]
```diff
@@ -0,0 +1,70 @@
...
+## What to check before you send a PR to code review
+* You use conventional commit messages
+* Unit tests passes
+* Other automatic tests passes, given it is feasible to execute them before code review 
+* Naming conventions
+
+## What to check in code reviews
+* Functionality: Does the code behave as the PR author likely intended? Does the code behave as users would expect? If relevant, test the new feature manually on your local machine and try edge cases that might break the code.
+* Software design: Is the code well-designed and fitted to the surrounding architecture?
+* Complexity: Would another developer be able to easily understand and use the code?
```
Comment (soofstad) [inline docs/how-to/using-git.md:66]:
```suggestion
* Readability: Would another developer be able to easily understand and work with the code? Are there none-idiomatic patterns?
```

**Review by soofstad — CHANGES_REQUESTED** [review]

File: docs/how-to/using-git.md [hunk 6]
```diff
@@ -0,0 +1,70 @@
...
+* test: adding tests, refactoring test; no production code change
+* chore: updating build tasks, package manager configs, etc; no production code change
+* perf: a code change that improves performance
+
+## Commits
+Keep commit messages informative.
+Here is a template for commit messages:
+
+```
+<type>(<optional scope>): imperative tense summary, <= 50 chars
```
Comment (mariuswinger) [inline docs/how-to/using-git.md:45]: What does <optional scope> refer to in this context?

File: docs/how-to/using-git.md [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+When necessarry, mote details can come here, until 72 chars each line.
```
Comment (mariuswinger) [inline docs/how-to/using-git.md:47]: typo: necessary. Rephrase: "When necessary" -> "If necessary"; "until 72" -> "up to 72 on each line"

File: docs/how-to/using-git.md [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+Refs/Resolves #<issue id>
+
+BREAKING CHANGE: <description>
+```
+
+Also, remember to set up [Pre-commit](pre-commit.md).
+
+
+## What to check before you send a PR to code review
+* You use conventional commit messages
```
Comment (mariuswinger) [inline docs/how-to/using-git.md:58]: The commit messages above are not described as "conventional" meaning that the word conventional in this list loses meaning unless you already know what it means. I suggest adding the word conventional in the paragraph above: "Commit messages should follow the conventional commit message specification". We could also consider linking to the documentation of conventional commit there-

File: docs/how-to/using-git.md [hunk 9]
```diff
@@ -0,0 +1,70 @@
...
+* Other automatic tests passes, given it is feasible to execute them before code review 
+* Naming conventions
+
+## What to check in code reviews
+* Functionality: Does the code behave as the PR author likely intended? Does the code behave as users would expect? If relevant, test the new feature manually on your local machine and try edge cases that might break the code.
+* Software design: Is the code well-designed and fitted to the surrounding architecture?
+* Complexity: Would another developer be able to easily understand and use the code?
+* Tests: Does the PR have correct and well-designed automated tests?
+* Naming: Are names for variables, functions, etc. descriptive?
+* Comments: Are the comments clear and useful?
```
Comment (mariuswinger) [inline docs/how-to/using-git.md:69]:
```suggestion
* Comments: Are comments understandable and useful?
```

File: docs/how-to/using-git.md [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+
+Using Git
+=============
+
+Git is used as version control system for the source code in the project.
+
+
+
+## Basic principles
+Whenever the devleoper makes any code changes, a new Pull Request (PR) should be created. The new PR should branch out from the main branch.
```
Comment (mariuswinger) [inline docs/how-to/using-git.md:13]:
```suggestion
Whenever a developer makes changes to the code, a new Pull Request (PR) should be created. The new PR should branch out from the main branch.
```

File: docs/how-to/using-git.md (hunk 6 above)
Comment (mariuswinger) [reply docs/how-to/using-git.md:45]: Also, it is not evident if type refers to the same types as above, or if this type parameter includes more or less types than above.

File: docs/how-to/using-git.md [hunk 11]
```diff
@@ -0,0 +1,70 @@
...
+* feat: a feature
+* fix: a bugfix
+* build: a change to build system or external dependencies
+* ci: a change to continous integration/delivery setup (typically scripts and config - files)
+* docs: a changes to documentation
+* style: a change to style 
+* refactor: a code change that neither fixes a bug nor adds/removes a feature
+* test: adding tests, refactoring test; no production code change
+* chore: updating build tasks, package manager configs, etc; no production code change
+* perf: a code change that improves performance
```
Comment (olavis) [inline docs/how-to/using-git.md:38]: Do we need all these types? Only `fix` and `feat` are mandated by the conventional commits specification. I've only found half of these types useful. If the type isn't a fix or feat I always find the prefix the least useful part of the commit message/branch name. Am I the only one?

File: docs/how-to/using-git.md [hunk 12]
```diff
@@ -0,0 +1,70 @@
...
+
+## Commits
+Keep commit messages informative.
```
Comment (olavis) [inline docs/how-to/using-git.md:41]: Maybe we can add some examples or do's/dont's? For instance messages like: `fixed comments from review` may seem informative within the scope of the PR, but are not really useful when you need to debug or go through the git history later.

File: docs/how-to/using-git.md [hunk 13]
```diff
@@ -0,0 +1,70 @@
...
+
+Refs/Resolves #<issue id>
+
+BREAKING CHANGE: <description>
+```
+
+Also, remember to set up [Pre-commit](pre-commit.md).
+
+
+## What to check before you send a PR to code review
```
Comment (olavis) [inline docs/how-to/using-git.md:57]: This section and `What to check in code reviews` should be merged. The team share equal responsibility for the code, and we should review our own code before asking team members.

File: docs/how-to/using-git.md [hunk 14]
```diff
@@ -0,0 +1,70 @@
...
+* Unit tests passes
+* Other automatic tests passes, given it is feasible to execute them before code review 
+* Naming conventions
+
+## What to check in code reviews
+* Functionality: Does the code behave as the PR author likely intended? Does the code behave as users would expect? If relevant, test the new feature manually on your local machine and try edge cases that might break the code.
+* Software design: Is the code well-designed and fitted to the surrounding architecture?
+* Complexity: Would another developer be able to easily understand and use the code?
+* Tests: Does the PR have correct and well-designed automated tests?
+* Naming: Are names for variables, functions, etc. descriptive?
```
Comment (olavis) [inline docs/how-to/using-git.md:68]: + following correct naming conventions, casing, aim for names to be _exact_ - makes it easier to avoid violation of the the single responsibility principle.

File: docs/how-to/using-git.md [hunk 15]
```diff
@@ -0,0 +1,70 @@
...
+Also, remember to set up [Pre-commit](pre-commit.md).
+
+
+## What to check before you send a PR to code review
+* You use conventional commit messages
+* Unit tests passes
+* Other automatic tests passes, given it is feasible to execute them before code review 
+* Naming conventions
+
+## What to check in code reviews
```
Comment (olavis) [inline docs/how-to/using-git.md:63]: Maybe we can list the SOLID principles, to have something thats more easily measured than `is the code well-designed?`

File: docs/how-to/using-git.md (hunk 10 above)
Comment (olavis) [inline docs/how-to/using-git.md:13]: developer

Comment (olavis) [reply docs/how-to/using-git.md:13]: Also, not sure if I agree that a new PR should always branch out from main. It can be very useful to branch out separate issues from a topic branch and create a PR into the given branch instead of just making a single (huge) PR that contains multiple features/fixes.

File: docs/getting-started/inital-setup.md [hunk 16]
```diff
@@ -85,3 +85,34 @@ To let the IDE know that dependencies can be found in `.yarn` instead of `node_m
...
+### Python
+1) Open settings in PyCharm
+2) Open Python interpreter
+3) Click on cogwheel and "Add..."
+4) Select Docker compose in the left menu and Choose the following settings:
+   1) Server: Docker
+   2) Configuration files: select docker-compose.yml and docker-compose.override.yml
+   3) Service: api
+   Click "OK" and "Apply"
+5) Click on "Add Configutarion..." in top right corner in PyCharm
```
Comment (olavis) [inline docs/getting-started/inital-setup.md:102]: Configuration

File: docs/how-to/using-git.md [hunk 17]
```diff
@@ -0,0 +1,70 @@
...
+
+
+## Basic principles
+Whenever the devleoper makes any code changes, a new Pull Request (PR) should be created. The new PR should branch out from the main branch.
+Ideally, each PR will do one thing (one new feature, fix one bug, etc). Keeping PRs small will simplify the code review process significantly. Also, a PR should be connected to an active issue from the project board.
+
+
+Tip: on Github, if you write the syntax "closes #XXX" (XXX is the issue number) in the PR description, the connected issue will be automatically closed when the PR is merged.
+
+When a branch (feature) is done, the developers create a PR and asks for review. When a PR is approved, the branch is merged to main. At least one person should approve the PR before merging to main. Merging will trigger the [github actions workflow](github-actions.md) which automatically tests and deploys.
```
Comment (olavis) [inline docs/how-to/using-git.md:19]: `When a PR is approved` What if the PR isn't approved? Or do we take it for granted? Ideally we should have some guidelines for when to close a PR, if it's growing too complex or has gotten outdated due to staying open for too long. A PR can affect the effectiveness of the team badly if it remains open.

**Comment by KristianKjerstad** [comment]
todo
should add info about env vars to docs (see for example https://github.com/equinor/QuatoOps/blob/master/docs/getting_started.md )

**Comment by KristianKjerstad** [comment]
note: some github repos add a description of the project file structure, which can be useful but also difficult to maintain

https://github.com/microsoft/MLOpsPython/blob/master/docs/code_description.md

File: docs/getting-started/prerequisites.md [hunk 18]
```diff
@@ -10,5 +10,10 @@ In order to run you need to have installed:
 - [Docker](https://www.docker.com/)
 - [Docker Compose](https://docs.docker.com/compose/)
 - Git
+
+If you want to run locally, you also need:
```
Comment (soofstad) [inline docs/getting-started/prerequisites.md:14]:
```suggestion
If you want to run tests, pre-commit etc. outside of docker, you also need:
```

**Review by soofstad — APPROVED** [review]

**Review by sebastianvitterso — APPROVED** [review]

**Comment by KristianKjerstad** [comment]
outdated, see #78
